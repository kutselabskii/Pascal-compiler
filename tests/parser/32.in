program abc;

var a: real;

begin
if (a > 0) then
	while (a > 0) do
		a := a / 2;
end.