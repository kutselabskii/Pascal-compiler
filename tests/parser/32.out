|__ program
    |-- abc //Program_heading
    |__ //Block
        |-- //Declaration part
        |   |-- var
        |   |   |-- a
        |   |   |   |__  : real
        |__ //Statement part
            |__ //Statement sequence
                |-- if
                |   |-- strictly_greater_than
                |   |   |-- a
                |   |   |   |__  : real
                |   |   |__ //Typecast
                |   |       |--  : real
                |   |       |__ 0
                |   |-- while
                |   |   |-- strictly_greater_than
                |   |   |   |-- a
                |   |   |   |   |__  : real
                |   |   |   |__ //Typecast
                |   |   |       |--  : real
                |   |   |       |__ 0
                |   |   |__ //Assignment_statement
                |   |       |-- a
                |   |       |   |__  : real
                |   |       |__ division
                |   |           |-- a
                |   |           |   |__  : real
                |   |           |__ //Typecast
                |   |               |--  : real
                |   |               |__ 2
